adapta/utils/python_typing/_functions.py

```diff
@@ -1,13 +1,24 @@
 """Common python typing functions. All of these are imported into __init__.py"""
-
+import sys
 from typing import Type, get_origin, Union, get_args
 
+TypeHint = Type
+if sys.version_info >= (3, 10):
+    from types import UnionType
+
+    TypeHint = Union[UnionType, Type]  # pylint: disable=invalid-name
```

Comment:
Perhaps it would be better to rename the `TypeHint` type rather than suppress the warning.

Reply:
Now called `ArgumentType` instead. Initially did not understand what the warning would accept. Apparently, type aliases cannot start with `T` or `Type`